analyzers/src/SonarAnalyzer.Common/Rules/ClassShouldNotBeEmptyBase.cs

```diff
@@ -44,15 +51,22 @@ public abstract class ClassShouldNotBeEmptyBase<TSyntaxKind> : SonarDiagnosticAn
             {
                 if (Language.Syntax.NodeIdentifier(c.Node) is { IsMissing: false } identifier
                     && IsEmptyAndNotPartial(c.Node)
-                    && !ShouldIgnoreBecauseOfBaseClass(c.Node, c.SemanticModel))
+                    && !HasAnyAttribute(c.Node)
+                    && !HasConditionalCompilationDirectives(c.Node)
+                    && !ShouldIgnoreBecauseOfBaseClassOrInterface(c.Node, c.SemanticModel))
                 {
                     c.ReportIssue(Diagnostic.Create(Rule, identifier.GetLocation(), DeclarationTypeKeyword(c.Node)));
                 }
             },
             Language.SyntaxKind.ClassAndRecordClassDeclarations);
 
-    private bool ShouldIgnoreBecauseOfBaseClass(SyntaxNode node, SemanticModel model) =>
+    private bool ShouldIgnoreBecauseOfBaseClassOrInterface(SyntaxNode node, SemanticModel model) =>
         IsClassWithDeclaredBaseClass(node)
-        && model.GetDeclaredSymbol(node) is INamedTypeSymbol classSymbol
-        && (classSymbol.BaseType is { IsAbstract: true } || classSymbol.DerivesFromAny(BaseClassesToIgnore));
+        && (HasGenericBaseClassOrInterface(node) || ShouldIgnoreType(node, model));
+
+    private static bool ShouldIgnoreType(SyntaxNode node, SemanticModel model) =>
+        model.GetDeclaredSymbol(node) is INamedTypeSymbol classSymbol
+            && (classSymbol.BaseType is { IsAbstract: true }
```

Comment:
[my education] could you remind me please why we ignore also subclasses of abstract base classes?

Reply:
If an abstract class:
- has no abstract methods: this is often used to provide the derived class a way to use the default implementation without adding any methods ([see this previous comment](https://github.com/SonarSource/sonar-dotnet/pull/6754#discussion_r1105918260))
- has abstract methods: if the class is empty it will result in a compilation error for not overriding them, so the analyzer can ignore this scenario